### PR TITLE
DST Integration

### PR DESCRIPTION
**Pull request by akhil-ta**

### CODE REVIEW

File: android/src/main/java/com/dieam/reactnativepushnotification/dst/DSTManager.java [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
+    private String DST_STATUS = DST.DST_START.name();
+    //private boolean timeZoneChangeReceiverIsRegistered = false;
+
+
+    public void scheduleDSTTransitionAlarms(Context context) {
+        simpleDateFormat = new SimpleDateFormat("dd/MMM/yyyy HH:mm a", Locale.getDefault());
+        alarmMgr = (AlarmManager) context.getSystemService(Context.ALARM_SERVICE);
+        List<DSTData> dstList = checkTimeZoneIsOnDSTAndGetDSTData(context, TimeZone.getDefault().getID());
+        Calendar currentCalendar = Calendar.getInstance();
+        /* Receiver should be registered once*/
```
Comment (kamalkkishoreta) [inline android/src/main/java/com/dieam/reactnativepushnotification/dst/DSTManager.java:41]: Remove commented code.

File: android/src/main/java/com/dieam/reactnativepushnotification/dst/DSTManager.java [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+            } else {
+                startDay = dstObject.getStart_day();
+                endDay = dstObject.getEnd_day();
+            }
+
+            Calendar startCalendar = Calendar.getInstance();
+            startCalendar.set(Calendar.MONTH, startMonth);
+            startCalendar.set(Calendar.DAY_OF_MONTH, startDay);
+            startCalendar.set(Calendar.HOUR_OF_DAY, startTime);
+            startCalendar.set(Calendar.MINUTE, 0);
```
Comment (kamalkkishoreta) [inline android/src/main/java/com/dieam/reactnativepushnotification/dst/DSTManager.java:72]: We should include minutes along with hour.

File: android/src/main/java/com/dieam/reactnativepushnotification/dst/DSTManager.java [hunk 3]
```diff
@@ -0,0 +1,392 @@
...
+            startCalendar.set(Calendar.SECOND, 0);
+            startCalendar.set(Calendar.MILLISECOND, 0);
+
+            Calendar endCalendar = Calendar.getInstance();
+            endCalendar.set(Calendar.MONTH, endMonth);
+            endCalendar.set(Calendar.DAY_OF_MONTH, endDay);
+            endCalendar.set(Calendar.HOUR_OF_DAY, endTime);
+            endCalendar.set(Calendar.MINUTE, 0);
```
Comment (kamalkkishoreta) [inline android/src/main/java/com/dieam/reactnativepushnotification/dst/DSTManager.java:80]: We should include minutes along with hour.

File: android/src/main/java/com/dieam/reactnativepushnotification/dst/DSTManager.java (hunk 1 above)
Comment (akhil-ta) [reply android/src/main/java/com/dieam/reactnativepushnotification/dst/DSTManager.java:41]: Will remove all unwanted logs after retesting. Will create another pull request for that.

File: android/src/main/java/com/dieam/reactnativepushnotification/dst/DSTManager.java (hunk 2 above)
Comment (akhil-ta) [reply android/src/main/java/com/dieam/reactnativepushnotification/dst/DSTManager.java:72]: Ok.

File: android/src/main/java/com/dieam/reactnativepushnotification/dst/DSTManager.java (hunk 3 above)
Comment (akhil-ta) [reply android/src/main/java/com/dieam/reactnativepushnotification/dst/DSTManager.java:80]: Ok noted.

File: android/src/main/java/com/dieam/reactnativepushnotification/dst/DSTManager.java (hunk 2 above)
Comment (akhil-ta) [reply android/src/main/java/com/dieam/reactnativepushnotification/dst/DSTManager.java:72]: Those are from Database value. The database only contains hour value.

File: android/src/main/java/com/dieam/reactnativepushnotification/dst/DSTManager.java (hunk 3 above)
Comment (akhil-ta) [reply android/src/main/java/com/dieam/reactnativepushnotification/dst/DSTManager.java:80]: Those are from Database value. The database only contains hour value.

**Pull request closed without merging**